### ISSUE TITLE
dotnet CLI crashing on Arch Linux

### ISSUE THREAD

**Issue by livarcocc** [dotnet/core-setup#3845]
_From @MaxXor on March 16, 2018 18:6_

## Steps to reproduce
1. Install `sudo pacman -S dotnet-sdk`
2. Create example console app `dotnet new console -o myApp`
3. Switch directory `cd myApp`
4. Run `dotnet run` (same applies for `dotnet build|restore`)

## Expected  behavior
Print `Hello World!`

## Actual behavior
```
Permission denied to modify the '/opt/dotnet/sdk/NuGetFallbackFolder' folder.

Here are some options to fix this error:

1. Re-run this command with elevated access.
2. Disabled the first run experience by setting the environment variable DOTNET_SKIP_FIRST_TIME_EXPERIENCE to true.
3. Copy the .NET Core SDK to a non-protected location and use it from there.
    
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.InvalidOperationException: The terminfo database is invalid.
   at System.TermInfo.Database..ctor(String term, Byte[] data)
   at System.TermInfo.Database.ReadDatabase(String term, String directoryPath)
   at System.TermInfo.Database.ReadDatabase(String term)
   at System.TermInfo.Database.ReadActiveDatabase()
   at System.ConsolePal.TerminalFormatStrings.<>c.<.cctor>b__27_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.ConsolePal.EnsureInitializedCore()
   at System.ConsolePal.ControlCHandlerRegistrar.Register()
   at System.Console.add_CancelKeyPress(ConsoleCancelEventHandler value)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine) in E:\A\_work\82\s\src\MSBuild\XMake.cs:line 527

Unhandled Exception: System.InvalidOperationException: The terminfo database is invalid.
   at System.TermInfo.Database..ctor(String term, Byte[] data)
   at System.TermInfo.Database.ReadDatabase(String term, String directoryPath)
   at System.TermInfo.Database.ReadDatabase(String term)
   at System.TermInfo.Database.ReadActiveDatabase()
   at System.ConsolePal.TerminalFormatStrings.<>c.<.cctor>b__27_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.ConsolePal.EnsureInitializedCore()
   at System.ConsolePal.ControlCHandlerRegistrar.Register()
   at System.Console.add_CancelKeyPress(ConsoleCancelEventHandler value)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine) in E:\A\_work\82\s\src\MSBuild\XMake.cs:line 755
   at Microsoft.Build.CommandLine.MSBuildApp.Main(String[] args) in E:\A\_work\82\s\src\MSBuild\XMake.cs:line 216


Unhandled Exception: System.InvalidOperationException: The terminfo database is invalid.
   at System.TermInfo.Database..ctor(String term, Byte[] data)
   at System.TermInfo.Database.ReadDatabase(String term, String directoryPath)
   at System.TermInfo.Database.ReadDatabase(String term)
   at System.TermInfo.Database.ReadActiveDatabase()
   at System.ConsolePal.TerminalFormatStrings.<>c.<.cctor>b__27_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.ConsolePal.WriteResetColorString()
   at System.ConsolePal.RefreshColors(ConsoleColor& toChange, ConsoleColor value)
   at System.Console.set_ForegroundColor(ConsoleColor value)
   at Microsoft.DotNet.Cli.Utils.AnsiConsole.SetColor(ConsoleColor color)
   at Microsoft.DotNet.Cli.Utils.AnsiConsole.Write(String message)
   at Microsoft.DotNet.Cli.Utils.AnsiConsole.WriteLine(String message)
   at Microsoft.DotNet.Cli.Utils.Reporter.WriteLine(String message)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
Aborted (core dumped)
```

## Environment data
```
.NET Command Line Tools (2.1.101)

Product Information:
 Version:            2.1.101
 Commit SHA-1 hash:  6c22303bf0

Runtime Environment:
 OS Name:     arch
 OS Version:  
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /opt/dotnet/sdk/2.1.101/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.6
  Build    : 74b1c703813c8910df5b96f304b0f2b78cdf194d
```

_Copied from original issue: dotnet/cli#8820_

**Comment by livarcocc** [comment]
This seems like a missing native dependency. Moving it to core-setup where we can get an answer faster.

**Comment by omajid** [comment]
Pretty sure this is https://github.com/dotnet/corefx/issues/26966

**Comment by MaxXor** [comment]
@omajid But it still tries to modify the NuGet backup folder when it should just use my local folder instead located at `~/.nuget/`. Is this not the problem?

> Permission denied to modify the '/opt/dotnet/sdk/NuGetFallbackFolder' folder.

**Comment by omajid** [comment]
Hmm. Good point. Maybe. I say wait for the terminfo stuff to be sorted out and retry.

You could also try following one of the suggestions it shows to see if that helps.

**Comment by PavelSrgv** [comment]
In Manjaro Linux in Konsole (from KDE) terminal, "dotnet new console" fails with similar error:
```
Unhandled Exception: System.InvalidOperationException: The terminfo database is invalid.
   at System.TermInfo.Database..ctor(String term, Byte[] data)
   at System.TermInfo.Database.ReadDatabase(String term, String directoryPath)
   at System.TermInfo.Database.ReadDatabase(String term)
   at System.TermInfo.Database.ReadActiveDatabase()
   at System.ConsolePal.TerminalFormatStrings.<>c.<.cctor>b__27_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Lazy`1.CreateValue()
   at System.ConsolePal.WriteResetColorString()
   at System.ConsolePal.RefreshColors(ConsoleColor& toChange, ConsoleColor value)
   at System.Console.set_ForegroundColor(ConsoleColor value)
   at Microsoft.DotNet.Cli.Utils.AnsiConsole.SetColor(ConsoleColor color)
   at Microsoft.DotNet.Cli.Utils.AnsiConsole.Write(String message)
   at Microsoft.DotNet.Cli.Utils.AnsiConsole.WriteLine(String message)
   at Microsoft.DotNet.Cli.Utils.Reporter.WriteLine(String message)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
Aborted (core dumped)
```
But it works fine in "xterm" terminal enulator. So, I use it as a workaround.

**Comment by Petermarcu** [comment]
@stephentoub any thoughts?

**Comment by stephentoub** [comment]
@stephentoub any thoughts?

Looks like https://github.com/dotnet/corefx/issues/26966.

**Comment by Petermarcu** [comment]
That was going to be my guess too :) Thats fixed and should be in the next wave of releases.

**Comment by livarcocc** [comment]
Closing as a dup of the corefx issue.

**Comment by trampster** [comment]
This started happening to my in the last week after updating my ubuntu on two different machines,

Permission denied to modify the '/usr/share/dotnet/sdk/NuGetFallbackFolder' folder.

Here are some options to fix this error:
----------------------------------------
1. Re-run this command with elevated access.
2. Disabled the first run experience by setting the environment variable DOTNET_SKIP_FIRST_TIME_EXPERIENCE to true.
3. Copy the .NET Core SDK to a non-protected location and use it from there.

I followed the standard guide for installing and didn't configure anything, and this error didn't happen until I updated. IE this appears to be the out of box experience after updating.
